docs/developer_guide.md

```diff
@@ -1,56 +1,92 @@
 # Developer Guide
 Thank you for contributing to RepoSense!
-## Architecture
+1. [Setting Up](#1-setting-up)
+2. [Architecture](#2-architecture)
+3. [Notable POJOs](#3-notable-pojos)
+4. [Frontend Components](#4-frontend-components)
+5. [Dashboard Generator Components](#5-dashboard-generator-components)
+6. [Git Utilities](#6-git-utilities)
+7. [Analyzer Components](#7-analyzer-components)
+8. [HTML Dashboard](#8-html-dashboard)
+
+## 1. Setting up
+
+### 1.1. Prerequisites
+
+1. **JDK `1.8.0_60`**  or later<br>
+> This app only works with Java 8 version. <br>
+
+2. **IntelliJ** IDE
+3. **Git** on the command line
+> Ensure that you're able to use it on the OS terminal. <br>
+
+### 1.2. Setting up the project in your computer
```

Comment:
Quick comment: may be not use section numbers? Markdown cannot generate section numbers automatically. Hard-coding them means it is hard to insert/delete sections later.

Reply:
hmm, true. In that case, shall we port over to asciidoc while its still in the early stage?